pkg/utils/helper.go

```diff
@@ -11,6 +12,64 @@ import (
 	"strings"
 )
 
+// PublicAzureStorageEndpointSuffix defines default Storage Endpoint Suffix
+const PublicAzureStorageEndpointSuffix = "core.windows.net"
+
+// Azure defines Azure configuration
+type Azure struct {
+	Cloud string `json:"cloud"`
+}
+
+// AzureStackCloud defines Azure Stack Cloud configuration
+type AzureStackCloud struct {
+	StorageEndpointSuffix string `json:"storageEndpointSuffix"`
+}
+
+// IsAzureStackCloud returns true if the application is running on Azure Stack Cloud
+func IsAzureStackCloud() bool {
+	azureFile, err := RunCommandOnHost("cat", "/etc/kubernetes/azure.json")
+	if err != nil {
+		return false
+	}
+	var azure Azure
+	err = json.Unmarshal([]byte(azureFile), &azure)
+	if err != nil {
+		return false
+	}
+	cloud := azure.Cloud
+	return strings.ToUpper(cloud) == "AZURESTACKCLOUD"
+}
+
+// CopyCert saves the cert content to a new file inside the container
+func CopyCert() error {
+	cert, err := RunCommandOnHost("cat", "/etc/ssl/certs/azsCertificate.pem")
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve cert content: %+v", err)

```

Comment:
Errors can be concatenate to each others, for instance the final prompt could be in the format `Error: error1: error2: ...` so starting errors message with a lower case is to me a better option + we know we are in an error, so I tend to avoid words such as "error" or "failed".
Finally, there is now error wrapping with `%w` so we should use it instead of `%v`
Example:
```go
if err != nil {
    // Return an error which unwraps to err.
    return fmt.Errorf("decompress %v: %w", name, err)
}
```